rust/parquet/src/record/mod.rs

```diff
@@ -16,9 +16,209 @@
 // under the License.
 
 //! Contains record-based API for reading Parquet files.
+//!
+//! Example usage of reading data untyped:
+//!
+//! ```rust,no_run
+//! use std::fs::File;
+//! use std::path::Path;
+//! use parquet::file::reader::{FileReader, SerializedFileReader};
+//! use parquet::record::types::Row;
+//!
+//! let file = File::open(&Path::new("/path/to/file")).unwrap();
+//! let reader = SerializedFileReader::new(file).unwrap();
+//! let iter = reader.get_row_iter::<Row>(None).unwrap();
+//! for record in iter.map(Result::unwrap) {
+//!     println!("{:?}", record);
+//! }
+//! ```
+//!
+//! Example usage of reading data strongly-typed:
+//!
+//! ```rust,no_run
+//! use std::fs::File;
+//! use std::path::Path;
+//! use parquet::file::reader::{FileReader, SerializedFileReader};
+//! use parquet::record::{Record, types::Timestamp};
+//!
+//! #[derive(Record, Debug)]
+//! struct MyRow {
+//!     id: u64,
+//!     time: Timestamp,
+//!     event: String,
+//! }
+//!
+//! let file = File::open(&Path::new("/path/to/file")).unwrap();
+//! let reader = SerializedFileReader::new(file).unwrap();
+//! let iter = reader.get_row_iter::<MyRow>(None).unwrap();
+//! for record in iter.map(Result::unwrap) {
+//!     println!("{}: {}", record.time, record.event);
+//! }
+//! ```
 
-mod api;
-pub mod reader;
+mod display;
+mod reader;
+mod schemas;
 mod triplet;
+pub mod types;
 
-pub use self::api::{List, ListAccessor, Map, MapAccessor, Row, RowAccessor};
+use std::{
+    collections::HashMap,
+    fmt::{self, Debug},
+};
+
+use crate::{
+    basic::Repetition,
+    column::reader::ColumnReader,
+    errors::Result,
+    schema::types::{ColumnPath, Type},
+};
+
+#[doc(inline)]
+pub use parquet_derive::Record;
+pub use reader::RowIter;
+pub use schemas::RootSchema;
+#[doc(hidden)]
+pub mod _private {
+    /// This is used by `#[derive(Record)]`
+    pub use super::display::DisplaySchemaGroup;
+}
+mod predicate {
```

Comment:
Do we need this now? can we add them in future when PPD is implemented?

Reply:
This is intended to avoid two breaking changes to the `get_row_iter<T>` methods – removal of the argument; and then re-adding it when dynamic projection and predicate pushdown are implemented. I don't mind either way, but I dislike API churn as much as anyone so that was my thinking!

Reply:
I'd prefer to remove it for now. We don't yet know what the API will look like - it can be defined in a backward compatible way.

Reply:
Done